components/tidb_query/src/codec/mysql/json/json_depth.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::Json;
+
+impl Json {
+    pub fn depth(&self) -> i64 {
+        match &self {
+            Json::Object(_) | Json::Array(_) => depth_json(self),
+            _ => 1,
+        }
+    }
+}
+
+pub fn depth_json(j: &Json) -> i64 {
+    let mut subdepth = 0;
+    let result = match *j {
+        Json::Array(ref array) => {
+            let mut result = 0;
+            for child in array {
+                let child_depth = depth_json(child);
+                if result < child_depth {
+                    result = child_depth;
+                }
+            }
+            result
+        }
+
+        Json::Object(ref map) => {
+            let mut result = 0;
+            for key in map.keys() {
+                let value_depth = depth_json(&map[key]);
+                if result < value_depth {
+                    result = value_depth;
+                }
+            }
+            result
+        }
+        _ => 0,
+    };
+    if result > subdepth {
```

Comment:
When will `result < subdepth`? Seems that if result >= subdepth, just return `result + 1` is ok.

Reply:
I think just :
```
match *j {
} + 1
```
It's same as your logic here.

Reply:
Done